central/processlisteningonport/store/postgres/full_store.go

```diff
@@ -49,11 +57,67 @@ func (s *fullStoreImpl) GetProcessListeningOnPort(
 		"ProcessListeningOnPortStorage",
 	)
 
+	// _, err := pgutils.Retry2(func() (pgx.Rows, error) {
+	//	return s.checkPermission(ctx, deploymentID)
+	// })
+
+	// if err != nil {
+	//	return nil, err
+	//}
```

Comment:
```suggestion
	extendedCtx := sac.WithAllAccess(ctx)
	deployment, err := s.deploymentStore.Get(extendedCtx, deploymentID)
	if err != nil { return nil, err }
	allowed, err := sac.ForResource(resources.Namespace).ReadAllowed(ctx, sac.KeyForNSScopedObj(deployment))
	if err != nil { return nil, err }
	if !allowed { return nil, nil }
```

Note: The deployment store is a cached one, which means lookup by ID will perform an internal map lookup (quick, without network access).
I'd avoid performing statements on tables that are not the ones storing the target types of the store layer within a store.

Scoped access control checks will return the same if an object does not exist in DB or if access to a namespace-scoped object such as deployment is denied. As a consequence, I'd rather return nothing if the deployment lookup did not return anything.

Reply:
The `deploymentStore` is internal, so I don't think I can use it directly, but I can still use `GetDeployment` in `deploymentDataStore`. However, I get a circular dependency.

```
$ go test -tags=sql_integration -v
package github.com/stackrox/rox/central/processlisteningonport/datastore
	imports github.com/stackrox/rox/central/processindicator/datastore
	imports github.com/stackrox/rox/central/processlisteningonport/store/postgres
	imports github.com/stackrox/rox/central/deployment/datastore
	imports github.com/stackrox/rox/central/processbaseline/datastore
	imports github.com/stackrox/rox/central/processindicator/datastore: import cycle not allowed
```

I have gone with the code in `checkAccess`.